client-flink/common/src/main/java/org/apache/celeborn/plugin/flink/readclient/FlinkShuffleClientImpl.java

```diff
@@ -565,6 +587,15 @@ public void setDataClientFactory(TransportClientFactory dataClientFactory) {
   @Override
   @VisibleForTesting
   public TransportClientFactory getDataClientFactory() {
+    initializeTransportClientFactoryIfRequired();
```

Comment:
`getDataClientFactory` gets invoked without `setupLifecycleManagerRef` primarily for tests.